music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,348 @@
+        '''
+
+        if not self.combinedList:
+            self.prepList()
+
+        f = fileName or self.fileName
+        text_file = open(os.path.join(outPath, f'{f}.txt'), "w")
+        for entry in self.combinedList:
+            text_file.write(entry + "\n")
+        text_file.close()
+
+
+# ------------------------------------------------------------------------------
+
+# Static functions
+
+def rnString(measure,
+             beat,
+             chordString: str,
+             inString: str = ''):
+    '''
+    Writes lines of RNTXT.
+    To start a new line: inString = None;
+    To extend an existing line: set inString to that existing list.
+    '''
+
+    if not inString:  # New line
+        inString = f'm{measure}'
+
+    bt = intBeat(beat)
+
+    newString = f'{inString} b{bt} {chordString}'
+
+    return newString
+
+
+def intBeat(beat,
+            roundValue: int = 2):
+    '''
+    Converts beats to integers if possible, and otherwise to rounded decimals.
+    Accepts input as string, int, float, or fractions.Fraction.
+    '''
+
+    options = [str, int, float, fractions.Fraction]
+
+    if type(beat) not in options:
+        raise TypeError(f'Beat, (currently {beat}) must be one of {options}.')
+
+    if isinstance(beat, int):
+        return beat
+
+    if type(beat) in [str, fractions.Fraction]:
+        beat = float(beat)
+
+    # Now beat is definitely a float
+    if int(beat) == beat:
+        return int(beat)
+    else:
+        return round(beat, roundValue)
+
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+    '''
+    Tests for both main analysis cases - one full, one partial - and for a template.
+    Additional test for smaller static functions.
+    '''
+
+    def testTwoCorpusPiecesAndTwoCorruptions(self):
+
+        from music21 import corpus
+        from music21 import meter
+
+        scoreBach = corpus.parse('bach/choraleAnalyses/riemenschneider001.rntxt',
+                                 format='RomanText')
+
+        rnaBach = RnWriter(scoreBach)
+        rnaBach.prepList()
+
+        self.assertEqual(rnaBach.combinedList[0], 'Composer: J. S. Bach')
+        self.assertEqual(rnaBach.combinedList[4], '\nTime Signature: 3/4')
+        self.assertEqual(rnaBach.combinedList[5], 'm0 b3 G: I')
+
+        # --------------------
+
+        scoreBach.parts[0].measure(3).insert(0, meter.TimeSignature('10/8'))
+        scoreBach.parts[0].measure(3).insert(1, meter.TimeSignature('5/8'))
+
+        wonkyBach = RnWriter(scoreBach)
+        wonkyBach.prepList()
+        self.assertEqual(wonkyBach.combinedList[8], '\nTime Signature: 10/8')
+        self.assertEqual(wonkyBach.combinedList[9],
+                         'Note: further time signature change(s) unprocessed: [\'5/8\']')
+
+        # --------------------
+
+        fullMonte = corpus.parse('monteverdi/madrigal.3.1.rntxt',
```

Comment:
similarly, too that this is likely too slow -- need timing.

Reply:
Replaced cases with smallest files in the Bach/Monteverdi collections. Now the test (9 of them including the new ones run in < 0.3 seconds). How's that? Is there a benchmark?